### DISCUSSION TITLE
sync_list ignored and no automatic sync

### DISCUSSION THREAD

**Discussion by DirkTurlach** [abraunegg/onedrive#2849] [Q&A] [answered]
Hey
I use v2.5.0-1+np2+2.1 on PopOS (Pop!_OS 22.04 LTS).

I use a sync_list and it is shown in display_config 
But it after onedrive --sync --resync -v
it downloads folders who are not in the List /Computer/Webseiten , /Computer/Ablage etc.
Whats wrong on my configuration?

The other problem is. 
If I performed a full resync (done before). There was no syncing. If I change a file in Onedrive on an other computer, these changings will not synchronised.

The resync needs about 20hours.

**onedrive --display-config** 
```
D-Bus message bus daemon is available; GUI notifications are now enabled
Reading configuration file: /home/dirk/.config/onedrive/config
Configuration file successfully loaded
Application version                          = onedrive v2.5.0-1+np2+2.1
Compiled with                                = LDC 2098
User Application Config path                 = /home/dirk/.config/onedrive
System Application Config path               = /etc/onedrive
Applicable Application 'config' location     = /home/dirk/.config/onedrive/config
Configuration file found in config location  = true - using 'config' file values to override application defaults
Applicable 'sync_list' location              = /home/dirk/.config/onedrive/sync_list
Applicable 'items.sqlite3' location          = /home/dirk/.config/onedrive/items.sqlite3
Config option 'drive_id'                     = 
Config option 'sync_dir'                     = ~/OneDrive
Config option 'enable_logging'               = false
Config option 'log_dir'                      = /var/log/onedrive
Config option 'disable_notifications'        = false
Config option 'skip_dir'                     = 
Config option 'skip_dir_strict_match'        = false
Config option 'skip_file'                    = ~*|.~*|*.tmp|*.swp|*.partial
Config option 'skip_dotfiles'                = false
Config option 'skip_symlinks'                = false
Config option 'monitor_interval'             = 300
Config option 'monitor_log_frequency'        = 12
Config option 'monitor_fullscan_frequency'   = 12
Config option 'read_only_auth_scope'         = false
Config option 'dry_run'                      = false
Config option 'upload_only'                  = false
Config option 'download_only'                = false
Config option 'local_first'                  = false
Config option 'check_nosync'                 = false
Config option 'check_nomount'                = false
Config option 'resync'                       = false
Config option 'resync_auth'                  = false
Config option 'cleanup_local_files'          = false
Config option 'classify_as_big_delete'       = 1000
Config option 'disable_upload_validation'    = false
Config option 'disable_download_validation'  = false
Config option 'bypass_data_preservation'     = false
Config option 'no_remote_delete'             = false
Config option 'remove_source_files'          = false
Config option 'sync_dir_permissions'         = 700
Config option 'sync_file_permissions'        = 600
Config option 'space_reservation'            = 52428800
Config option 'application_id'               = d50ca740-c83f-4d1b-b616-12c519384f0c
Config option 'azure_ad_endpoint'            = 
Config option 'azure_tenant_id'              = 
Config option 'user_agent'                   = ISV|abraunegg|OneDrive Client for Linux/v2.5.0-1+np2+2.1
Config option 'force_http_11'                = false
Config option 'debug_https'                  = false
Config option 'rate_limit'                   = 0
Config option 'operation_timeout'            = 3600
Config option 'dns_timeout'                  = 60
Config option 'connect_timeout'              = 10
Config option 'data_timeout'                 = 60
Config option 'ip_protocol_version'          = 1
Config option 'threads'                      = 8
Environment var 'XDG_RUNTIME_DIR'            = true
Environment var 'DBUS_SESSION_BUS_ADDRESS'   = true
Config option 'notify_file_actions'          = false

Selective sync 'sync_list' configured        = true
sync_list config option 'sync_root_files'    = false
sync_list contents:
Dokumente
Computer/Programmieren
texte_schreiben
DSS
Medien/Bilder/Eigene Aufnahmen
Medien/Bilder/Fotos-Auswahlen
Reisen
Poitik-Gesellschaft
Kreativ
Entwicklung

Config option 'sync_business_shared_items'   = false

Config option 'webhook_enabled'              = false
```

**config-file in ~/.config/onedrive/config**
```
 Configuration for OneDrive Linux Client
# This file contains the list of supported configuration fields
# with their default values.
# All values need to be enclosed in quotes
# When changing a config option below, remove the '#' from the start of the line
# For explanations of all config options below see docs/usage.md or the man page.
#
# sync_dir = "~/OneDrive"
# skip_file = "~*|.~*|*.tmp"
# monitor_interval = "300"
# skip_dir = ""
# log_dir = "/var/log/onedrive/"
# drive_id = ""
# upload_only = "false"
# check_nomount = "false"
# check_nosync = "false"
# download_only = "false"
# disable_notifications = "false"
# disable_upload_validation = "false"
# enable_logging = "false"
# force_http_11 = "false"
# local_first = "false"
# no_remote_delete = "false"
# skip_symlinks = "false"
# debug_https = "false"
# skip_dotfiles = "false"
# skip_size = "1000"
# dry_run = "false"
# min_notify_changes = "5"
# monitor_log_frequency = "6"
# monitor_fullscan_frequency = "12"
# sync_root_files = "false"
# classify_as_big_delete = "1000"
# user_agent = ""
# remove_source_files = "false"
# skip_dir_strict_match = "false"
# application_id = ""
# resync = "false"
# local_first = "false"
# resync = "false"
# resync_auth = "false"
# bypass_data_preservation = "false"
# azure_ad_endpoint = ""
# azure_tenant_id = "common"
# sync_business_shared_items = "false"
# sync_dir_permissions = "700"
# sync_file_permissions = "600"
# rate_limit = "131072"
# operation_timeout = "3600"
# webhook_enabled = "false"
# webhook_public_url = ""
# webhook_listening_host = ""
# webhook_listening_port = "8888"
# webhook_expiration_interval = "600"
# webhook_renewal_interval = "300"
# webhook_retry_interval = "60"
# space_reservation = "50"
# display_running_config = "false"
# read_only_auth_scope = "false"
# cleanup_local_files = "false"
ip_protocol_version = "1"
```

Any Idea?

Thanks Dirk

**Comment by abraunegg** [comment, marked as answer]
@DirkTurlach 
> sync_list ignored and no automatic sync

This is a regression as per https://github.com/abraunegg/onedrive/issues/2832 and is already fixed in 'master'.

> `sync_list contents:`

Please review your 'sync_list' contents as you are *searching* for those entries in *any* path - if that is truly what you want ... fine .. but that adds to the complexity of path scanning.

Please read the usage document on how to configure your 'sync_list' correctly:
* https://github.com/abraunegg/onedrive/blob/master/docs/usage.md#using-client-side-filtering-rules-to-determine-what-should-be-synced-with-microsoft-onedrive
* https://github.com/abraunegg/onedrive/blob/master/docs/usage.md#performing-a-selective-synchronisation-via-sync_list-file

> The resync needs about 20hours.

That is way too long for this to occur .... what hardware are you running this on, how much system memory do you have? How many items on OneDrive do you have? details on:
* CPU details
* Memory details
* Swap space details
* Local disk speed details
* Number of OneDrive items

As per the documentation (https://github.com/abraunegg/onedrive/blob/master/docs/usage.md#memory-usage):

Testing indicates that during a `--resync` operation or a full online scan, the client may use approximately 1GB of free memory for every 100,000 objects stored online. This occurs because the client queries the OneDrive API for all objects before processing them locally. Once processing is complete, the memory is released back to the system.

**Reply by DirkTurlach** [reply]
Hey, thanks for your answer.

Ich think, perhaps the / was missed at the Ende of line (Computer/Programmieren/  instead of Computer/Programmieren) ?
That would be an explanation.

What I realy want is only sync some directorys. 
My next first step would be to sync only one directory with all Content of Subdirectories. 
For that I need only one line in the sync_list ? For example : 

_MyDirectory_

If I understand it right, that should work?

The next step would be to sync only some subdirectorys of one dir:
For that it should look like:

_MyDirectory
MyDirectory2/sub1/
MyDirectory2/sub2/
MyDirectory2/sub3/_

Should that work?

___________________________________

My system is not so bad. Here are my configuration:
I will check my memory if try the next time

Betriebsystemname	Microsoft Windows 11 Home
Version	10.0.22631 Build 22631
Weitere Betriebsystembeschreibung 	Nicht verfügbar
Betriebsystemhersteller	Microsoft Corporation
Systemname	ROG
Systemhersteller	ASUSTeK COMPUTER INC.
Systemmodell	ROG Flow X13 GV301RE_GV301RE
Systemtyp	x64-basierter PC
System-SKU	
Prozessor	AMD Ryzen 9 6900HS with Radeon Graphics, 3301 MHz, 8 Kern(e), 16 logische(r) Prozessor(en)
BIOS-Version/-Datum	American Megatrends International, LLC. GV301RE.331, 29.03.2024
SMBIOS-Version	3.4
Version des eingebetteten Controllers	3.16
BIOS-Modus	UEFI
BaseBoard-Hersteller	ASUSTeK COMPUTER INC.
BaseBoard-Produkt	GV301RE
BaseBoard-Version	1.0
Plattformrolle	Mobil
Sicherer Startzustand	Aus
PCR7-Konfiguration	Erweiterung zum Anzeigen erforderlich
Windows-Verzeichnis	C:\WINDOWS
Systemverzeichnis	C:\WINDOWS\system32
Startgerät	\Device\HarddiskVolume1
Gebietsschema	Deutschland
Hardwareabstraktionsebene	Version = "10.0.22621.2506"
Benutzername	ROG\dirkt
Zeitzone	Mitteleuropäische Sommerzeit
Installierter physischer Speicher (RAM)	32,0 GB
Gesamter physischer Speicher	31,2 GB
Verfügbarer physischer Speicher	20,0 GB
Gesamter virtueller Speicher	33,2 GB
Verfügbarer virtueller Speicher	20,0 GB
Größe der Auslagerungsdatei	2,00 GB
Auslagerungsdatei	C:\pagefile.sys
Kernel-DMA-Schutz	Ein
Virtualisierungsbasierte Sicherheit	Wird ausgeführt...
Virtualisierungsbasierte Sicherheit – erforderliche Sicherheitseigenschaften	
Virtualisierungsbasierte Sicherheit – verfügbare Sicherheitseigenschaften	Allgemeine Virtualisierungsunterstützung, DMA-Schutz, UEFI-Code Readonly, Modusbasierte Ausführungssteuerung
Virtualisierungsbasierte Sicherheit – konfigurierte Dienste	Durch Hypervisor erzwungene Codeintegrität
Virtualisierungsbasierte Sicherheit – ausgeführte Dienste	Durch Hypervisor erzwungene Codeintegrität
Windows Defender-Anwendungssteuerungsrichtlinie	Erzwungen
Windows Defender-Anwendungssteuerungs-Richtlinie für den Benutzermodus	Aus
Unterstützung der Geräteverschlüsselung	Erweiterung zum Anzeigen erforderlich
Es wurde ein Hypervisor erkannt. Features, die für Hyper-V erforderlich sind, werden nicht angezeigt.	



________________________

**Comment by abraunegg** [comment]
@DirkTurlach 
> ```
> MyDirectory
> MyDirectory2/sub1/
> MyDirectory2/sub2/
> MyDirectory2/sub3/
> 
> Should that work?

No .... it should be:
```
/MyDirectory/
/MyDirectory2/sub1/
/MyDirectory2/sub2/
/MyDirectory2/sub3/
```

Please re-read the documentation for 'sync_list'

Example ... `/MyDirectory/` path equates/evaluates to `~/OneDrive/MyDirectory/`
